dora/core/common/src/main/java/alluxio/membership/WorkerServiceEntity.java

```diff
@@ -127,4 +126,24 @@ public void deserialize(byte[] buf) {
         .create();
     gson.fromJson(new InputStreamReader(new ByteArrayInputStream(buf)), WorkerServiceEntity.class);
   }
+
+  /**
+   * A customized equality comparison which uses the customized comparison of WorkerNetAddress
+   * objects.
+   *
+   * @param o The object to be compared with this WorkerServiceEntity for customized equality
+   * @return true if the specified object is equal to this WorkerServiceEntity; false otherwise
+   */
+  public boolean customizedEquals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    WorkerServiceEntity anotherO = (WorkerServiceEntity) o;
+    return mIdentity.equals(anotherO.mIdentity)
+        && getServiceEntityName().equals(anotherO.getServiceEntityName())
+        && mAddress.customizedEquals(anotherO.getWorkerNetAddress());
```

Comment:
move the comparison here

Reply:
use name as consideredEquivalent ?

Reply:
yeah, or be explicit about what is considered equivalent, e.g. `equalsIgnoringOptionalFields`

Reply:
Updated the function name to `equalsIgnoringOptionalFields`.